src/Makefile

```diff
@@ -26,7 +26,7 @@ endif
 CXXFLAGS=-O2 -g -Wall -std=c++11 
 override CXXFLAGS +=-Iinclude
 ifeq ($(STATIC), yes)
-LDFLAGS += -lssl -lcrypto -Wl,--whole-archive -lpthread -Wl,--no-whole-archive -ldl -static
+LDFLAGS += -lssl -lcrypto -Wl,--whole-archive -lpthread -Wl,--no-whole-archive -ldl -lz -static -s

```

Comment:
zlib is not necessary. In some toolchains, zlib may not exist, and this will cause compilation to fail.
Such as musl-based systems.

Reply:
ok